packages/client/src/monaco-languages.ts

```diff
@@ -46,20 +46,70 @@ export function testGlob(pattern: string, value: string): boolean {
     return regExp.test(value);
 }
 
-export class MonacoLanguages implements Languages {
+class ExtHostDiagnostics {
+    private static _idPool: number = 0;
+    private readonly _collections = new Map<string, DiagnosticCollection>();
+
+    constructor(
+        protected readonly _monaco: typeof monaco,
+        protected readonly p2m: ProtocolToMonacoConverter
+    ) {
+    }
+
+    createDiagnosticCollection(name?: string): DiagnosticCollection {
+        let owner: string;
+        if (!name) {
+            name = '_generated_diagnostic_collection_name_#' + ExtHostDiagnostics._idPool++;
+            owner = name;
+        } else if (!this._collections.has(name)) {
+            owner = name;
+        } else {
+            do {
+                owner = name + ExtHostDiagnostics._idPool++;
+            } while (this._collections.has(owner));
```

Comment:
How shall this terminate?

Reply:
As soon as the collection doesn't contains the generated owner

btw, this code is not mine but is pretty much copy/paster from [vscode](https://github.com/microsoft/vscode/blob/e9393b096d90a5ac49b1528d1ec60483fc6b993e/src/vs/workbench/api/common/extHostDiagnostics.ts#L278)

Reply:
Ok, that's where it is coming from.

Reply:
sure: 00cfc7f1c38a4738df1bcb6a441923f8b2014bea